src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -327,6 +332,12 @@ private static TypeSymbol NarrowObjectType(ITypeManager typeManager, ObjectSynta
                     positionable = declarationParent.Name;
                     blockName = declarationParent.Keyword.Text;
                 }
+                else if (parent is IfExpressionSyntax ifExpressionSyntax &&

```

Comment:
Should we factor out the entire if...elseif chain into its own method?

Reply:
This change is reverted now since the grammar is updated.